lib/rules/selector-max-type/__tests__/index.mjs

```diff
@@ -33,15 +33,6 @@ testRule({
 		{
 			code: '.foo { &.bar {} }',
 		},
-		{
-			code: '.foo { &-bar {} }',
-		},
-		{
-			code: '.foo { &__bar {} }',
-		},
-		{
-			code: '.foo { [&] {} }',
-		},
```

Comment:
This can not be parsed and gives a warning, but it also isn't a valid selector.

I would prefer to just drop this test because bogus inputs will always give bogus outputs.

Reply:
Don't know if our policies have changed since 

> [Non-standard selectors should be ignored, and tests for them should be accepted.](https://github.com/stylelint/stylelint/pull/2665#issuecomment-311105801)

but shouldn't it fail gracefully?

Reply:
Are non-standard selectors all selectors that aren't standard, or are they only those that are valid in sass, less,...

Nuance between :
- completely bogus
- valid in a CSS flavor, but not in standard CSS

Reply:
I don't know but I think that we should at least communicate with `stylelint-scss` so that they can prepare a rule synchronised with our next release to offer as a drop-in solution to annoyed users.
That way you may drop anything that isn't standard.

Reply:
I'm sorry, I think there is some confusion here :)

`[&]` is not a sass or less selector.

<img width="307" alt="Screenshot 2024-02-01 at 21 00 38" src="https://github.com/stylelint/stylelint/assets/11521496/d5e5ddda-3cd5-4b42-8e93-5247fd392c52">

It isn't a valid selector anywhere as far as I know.
If it is valid anywhere, please let me know because then my entire premise is incorrect.

`selector-max-type` already used and continues to use `isStandardSyntaxSelector` to ignore any selector that is detectable as sass, less, ...

That `[&]` was fine in this rule was a false negative, likely a result of `postcss-resolve-nested-selector` doing find/replace with strings.

Reply:
> Editted to clarify that this comment is only about [&]

Sorry my bad, I was commenting on the basis that it was about the lines removed from line 36 to 44 which contained.

`code: '.foo { &__bar {} }',`